apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -711,10 +712,22 @@ public Iterable<? extends List<Class<?>>> onError(int index, List<Class<?>> batc
             .or(nameContains("javassist"))
             .or(nameContains(".asm."))
             .or(anyMatch(coreConfiguration.getDefaultClassesExcludedFromInstrumentation()))
-            .or(anyMatch(coreConfiguration.getClassesExcludedFromInstrumentation()))
+            .or(anyMatch(sanitizeConfiguredExcludes(logger, coreConfiguration.getClassesExcludedFromInstrumentation())))
             .disableClassFormatChanges();
     }
 
+    private static List<WildcardMatcher> sanitizeConfiguredExcludes(Logger logger, List<WildcardMatcher> classesExcludedFromInstrumentation) {
```

Comment:
I'd move this into `CoreConfiguration::getClassesExcludedFromInstrumentation` so that there's no way to circumvent the sanitization.

Reply:
As discussed in slack, I also adopted the mechanism to log only once per config change in order to prevent log-spam.